### PR TITLE
Refactor S60x

### PR DESCRIPTION
**Pull request by mechmerlin**
## Description

Refactored S60-X to do the following:
1. Force users to select which version rgb, or default when compiling firmware to avoid flashing defaults onto non supported hardware. 
2. Introduce HHKB LAYOUT macro and QMK Configurator support
3. Have both rgb and default pcbs share one info.json file
4. Cleanups such as `#pragma once` usage

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sentraq/s60_x/default/config.h [hunk 1]
```diff
@@ -1,20 +1,47 @@
...
-#define DEFAULT_CONFIG_H
+#pragma once
 
 #include "config_common.h"
 
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Massdrop
```
Comment (noroadsleft) [inline keyboards/sentraq/s60_x/default/config.h]: I realize this is left over from current master, but shouldn't this be

```suggestion
#define MANUFACTURER    Sentraq
```

?

File: keyboards/sentraq/s60_x/default/config.h [hunk 2]
```diff
@@ -1,20 +1,47 @@
...
 #define PRODUCT         S60-X
 #define DESCRIPTION     q.m.k. keyboard firmware for S60-X
```
Comment (noroadsleft) [inline keyboards/sentraq/s60_x/default/config.h]: This bothers me.

```suggestion
#define DESCRIPTION     QMK keyboard firmware for S60-X
```

File: keyboards/sentraq/s60_x/default/rules.mk [hunk 3]
```diff
@@ -1,5 +1,68 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+MIDI_ENABLE = no            # MIDI controls
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+
+
+LAYOUTS = 60_ansi 60_ansi_split_bs_rshift 60_iso
```
Comment (noroadsleft) [inline keyboards/sentraq/s60_x/default/rules.mk]: You should be able to add `60_hhkb` here.

```suggestion
LAYOUTS = 60_ansi 60_ansi_split_bs_rshift 60_iso 60_hhkb
```

File: keyboards/sentraq/s60_x/rgb/config.h [hunk 4]
```diff
@@ -1,22 +1,43 @@
...
-#define RBG_CONFIG_H
+#pragma once
 
 #include "config_common.h"
 
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Massdrop
```
Comment (noroadsleft) [inline keyboards/sentraq/s60_x/rgb/config.h]:
```suggestion
#define MANUFACTURER    Sentraq
```

File: keyboards/sentraq/s60_x/rgb/config.h [hunk 5]
```diff
@@ -1,22 +1,43 @@
...
 #define PRODUCT         S60-X-RGB
 #define DESCRIPTION     q.m.k. keyboard firmware for S60-X RGB
```
Comment (noroadsleft) [inline keyboards/sentraq/s60_x/rgb/config.h]:
```suggestion
#define DESCRIPTION     QMK keyboard firmware for S60-X RGB
```

File: keyboards/sentraq/s60_x/rgb/rules.mk [hunk 6]
```diff
@@ -1,5 +1,73 @@
...
 CONSOLE_ENABLE = no        # Console for debug(+400)
 COMMAND_ENABLE = no        # Commands for debug and configuration
 NKRO_ENABLE = yes          # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
 BACKLIGHT_ENABLE = yes     # Enable keyboard backlight functionality
-RGBLIGHT_ENABLE = yes      # Enable RGB light
\ No newline at end of file
+RGBLIGHT_ENABLE = yes      # Enable RGB light
+
+
+LAYOUTS = 60_ansi 60_ansi_split_bs_rshift 60_iso
```
Comment (noroadsleft) [inline keyboards/sentraq/s60_x/rgb/rules.mk]:
```suggestion
LAYOUTS = 60_ansi 60_ansi_split_bs_rshift 60_iso 60_hhkb
```

File: keyboards/sentraq/s60_x/readme.md [hunk 7]
```diff
@@ -1,12 +1,12 @@
-S60-x
-=====
+# S60-x
```
Comment (noroadsleft) [inline keyboards/sentraq/s60_x/readme.md]: Suggestions for this file: https://github.com/qmk/qmk_firmware/commit/1446a2f2fbfb117aceb52842a6671fec8ac87418  
The edit to Line 13 is required, however. (Refactor causes `sentraq/s60_x` to no longer be a valid target.)

Either that, or fork this into two readmes, one for each version.

File: keyboards/sentraq/s60_x/rgb/rules.mk [hunk 8]
```diff
@@ -1,5 +1,73 @@
...
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+#   Teensy halfKay   512
+#   Teensy++ halfKay 1024
+#   Atmel DFU loader 4096
+#   LUFA bootloader  4096
+#   USBaspLoader     2048
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/sentraq/s60_x/rgb/rules.mk]: What's the bootloader?

**Comment by drashna** [comment]
Thanks!

**Pull request merged**